.github/ci_templates/e2e_tests.yml

```diff
@@ -356,11 +430,135 @@ check_e2e_labels:
+          ls -la ${TMP_DIR_PATH}
+          exit 1
+        else
+          echo "Create temporary dir for job: ${TMP_DIR_PATH}."
+          mkdir -p "${TMP_DIR_PATH}"
+        fi
+
+        INSTALL_IMAGE_NAME="${DECKHOUSE_REGISTRY_HOST:-}${INSTALL_IMAGE_PATH}"
+
+        SAFE_IMAGE_NAME=$(echo ${INSTALL_IMAGE_NAME} | tr '[:lower:]' '[:upper:]')
+        echo "Deckhouse Deployment will use install image ${SAFE_IMAGE_NAME} to test Git ref ${REF_FULL}"
+
+        # Print image name in uppercase to prevent hiding non-secret registry host stored in secret.
+        echo "⚓️ [$(date -u)] Pull 'dev/install' image '${SAFE_IMAGE_NAME}'."
+        docker pull "${INSTALL_IMAGE_NAME}"
+
+        arrPath=(${INSTALL_IMAGE_PATH//:/ })
+        DECKHOUSE_IMAGE_TAG="${arrPath[1]}"
+
+        echo '::echo::on'
+        echo "::set-output name=tmp-dir-path::${TMP_DIR_PATH}"
+        echo "::set-output name=install-image-full::${INSTALL_IMAGE_NAME}"
+        echo "::set-output name=deckhouse-image-tag::${DECKHOUSE_IMAGE_TAG}"
+        echo "::set-output name=initial-image-tag::${INITIAL_IMAGE_TAG}"
+
+        echo '::echo::off'
+
+    - name: "Download state"
+      id: download_artifact_with_state
+      uses: dawidd6/action-download-artifact@v2.23.0
+      with:
+        github_token: ${{secrets.BOATSWAIN_GITHUB_TOKEN}}
+        run_id: ${{github.event.inputs.run_id}}
+        name: ${{github.event.inputs.state_artifact_name}}
+        path: ${{ steps.setup.outputs.tmp-dir-path}}
+
+    - name: Cleanup bootstrapped cluster
+      if: ${{ success() }}
+      id: cleanup_cluster
+      env:
+        PROVIDER: {!{ $ctx.providerName }!}
+        CRI: {!{ $ctx.criName }!}
+        LAYOUT: {!{ $ctx.layout }!}
+        KUBERNETES_VERSION: "{!{ $ctx.kubernetesVersion }!}"
+        LAYOUT_DECKHOUSE_DOCKERCFG: ${{ secrets.LAYOUT_DECKHOUSE_DOCKERCFG }}
+        LAYOUT_SSH_KEY: ${{ secrets.LAYOUT_SSH_KEY}}
+        TMP_DIR_PATH: ${{ steps.setup.outputs.tmp-dir-path}}
+        PREFIX: ${{ github.event.inputs.cluster_prefix }}
+        INSTALL_IMAGE_NAME: ${{ steps.setup.outputs.install-image-full }}
+        DECKHOUSE_IMAGE_TAG: ${{ steps.setup.outputs.deckhouse-image-tag }}
+        MASTER_CONNECTION_STRING: ${{ github.event.inputs.ssh_connection_string }}
+  {!{- tmpl.Exec "e2e_run_template" (slice .provider "cleanup") | strings.Indent 6 }!}
+
+    - name: Remove failed cluster label
+      if: ${{ success() }}
+      uses: actions-ecosystem/action-remove-labels@v1
+      with:
+        github_token: ${{ secrets.BOATSWAIN_GITHUB_TOKEN }}
+        number: ${{ github.event.inputs.issue_number }}
+        labels: "e2e/cluster/failed"
+
+    - name: Cleanup temp directory
+      if: always()
+      env:
+        TMPPATH: ${{ steps.setup.outputs.tmppath}}
+      run: |
+        echo "Remove temporary directory '${TMPPATH}' ..."
+        if [[ -d "${TMPPATH}" && ${#TMPPATH} > 1 ]] ; then
+          rm -rf "${TMPPATH}"
+        else
+          echo Not a directory.
+        fi
```

Comment:
Just a TODO for the future:
There is a RUNNER_TEMP env with the exact scope: a temporary directory for the job run. It was a workaround for Gitlab to store rendered layouts in the separate directory. Now we use "actions/upload-artifact" to store layouts and logs.